sysfs/gpio.go

```diff
@@ -505,7 +505,75 @@ func (d *driverGPIO) parseGPIOChip(path string) error {
 	return nil
 }
 
+func getSymlinkRoot(pinNumber int) string {
+	// The NVidia Jetson Orin AGX uses nonstandard names within /sys/class/gpio. This is a mapping
+	// from pin numbers to their names on that machine.
+	jetsonOrinAgxPins := map[int]string{
+		316: "AA.00", 317: "AA.01", 318: "AA.02", 319: "AA.03", 320: "AA.04", 321: "AA.05",
```

Comment:
This code allocates the whole map at every single function call. This is inefficient. A slightly better version would look like this (simplified form):

```
var mapping[...]string{"AA.00", "AA.01", ...}

p -= 316
return mapping[i]
```

I have concerns though;

https://docs.nvidia.com/jetson/archives/r35.1/DeveloperGuide/text/HR/JetsonModuleAdaptationAndBringUp/JetsonAgxOrinSeries.html#changing-the-pinmux

and

https://docs.nvidia.com/jetson/archives/r35.1/DeveloperGuide/text/HR/JetsonModuleAdaptationAndBringUp/JetsonAgxOrinSeries.html#identifying-the-gpio-number

has two interesting bits:

- In the JetPack 5.0.2-GA and later, you can use the GPIO sysfs tool to update the pinmux register
- Because the Jetson module dynamically registers GPIOs, search the kernel messages to check the GPIO allocation ranges for each GPIO group

so basically, they say the pin mapping name can change?

Reply:
> the pin mapping name can change

Ugh... you're right, but fixing that so it Just Works for everyone seems very hard. To search the kernel messages, I ran `dmesg | grep gpiochip` on my Orin (per your second link), and discovered that I need to use `sudo` to use `dmesg`. but I really don't want to require root access to use this library, so the approach of "parse `sudo dmesg | grep gpio` to get the base numbers" isn't going to work in a straightforward way.

How would you feel about just putting a warning in the readme, that if you change the pinmux on an Orin, this library probably won't work right? My gut instinct is that almost no one will change the pinmux, and supporting that feature isn't worth the hassle, though I could be wrong. and getting something that works most of the time beats something that doesn't work at all, even if it's not perfect.


(I'm still working on the rest of your feedback, but should be done soon)